pkg/ast/common.go

```diff
@@ -37,10 +38,23 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type CaseConversionInfo struct {
+	dualCaseCheckEnabled   bool
+	valueIsCaseInSensitive bool
+	valueIsRegex           bool
+	IsString               bool
+	originalColValue       interface{}
+}
+
+func (cci *CaseConversionInfo) ShouldSearchWithOriginalCase() bool {
+	return cci.IsString && cci.dualCaseCheckEnabled && cci.valueIsCaseInSensitive && !cci.valueIsRegex
+}
+
 // When valueIsRegex is true, colValue should be a string containing the regex
 // to match and should not have quotation marks as the first and last character
 // unless those are intended to be matched.
-func ProcessSingleFilter(colName string, colValue interface{}, compOpr string, valueIsRegex bool, qid uint64) ([]*FilterCriteria, error) {
+// If shouldBeCaseSensitive is set to true, valueIsCaseInSensitive will be ignored
+func ProcessSingleFilter(colName string, colValue interface{}, originalColValue interface{}, compOpr string, valueIsRegex bool, valueIsCaseInSensitive bool, shouldBeCaseSensitive bool, qid uint64) ([]*FilterCriteria, error) {
```

Comment:
So the search is `insensitive` only for the query that comes from `SPL` and even in that for `Transaction` command I am using the same function to parse the search nodes and for that I need the search to be `case-sensitive`. 
And if in future if we add some other parsing definition in `SPL` that uses the same function for parsing search nodes but they don't want the search to be `case-insensitive`, then we should set `shouldBeCaseSensitive` to  `true` to override the default `case-insensitive`. 

I did this way, so that I had to override but also assign the `originalColValue` to `colValue`.  

And the reason why I chose `valueIsCaseInSensitive` across the PR (only here I have both sensitive and insensitive). It is because only when the `valueIsCaseInSensitive` set to true, I should do the `dualCaseCheck` (if it is enabled). And for other parsers like `SQL` or `PIPE QL`, the `valueIsCaseInSensitive` is set to false by default and that is how we want. 

If I use `sensitive` as my value, then I need to set it to true across other parsers and since this is only for `SPL` parser I chose the opposite way that we need to set `insensitive` to true.